scopes/scope/scope/scope.main.runtime.ts

```diff
@@ -351,8 +353,39 @@ export class ScopeMain implements ComponentFactory {
   private localAspects: string[] = [];
 
   async loadAspects(ids: string[], throwOnError = false): Promise<void> {
+    const scopeManifests = await this.getManifestsGraphRecursively(ids, throwOnError);
+    await this.aspectLoader.loadExtensionsByManifests(scopeManifests);
+  }
+
+  async getManifestsGraphRecursively(ids: string[], throwOnError = false): Promise<ManifestOrAspect[]> {
+    if (!ids.length) return [];
+    const components = await this.getNonLoadedAspects(ids);
+    const manifests = await this.requireAspects(components, throwOnError);
+    await mapSeries(manifests, async (manifest) => {
+      if (manifest.dependencies) {
+        const depIds = manifest.dependencies.map((d) => d.id).filter((id) => id);
+        const loaded = await this.getManifestsGraphRecursively(depIds, throwOnError);
+        manifests.push(...loaded);
+      }
+      // @ts-ignore
+      if (manifest._runtimes) {
+        // @ts-ignore
+        await mapSeries(manifest._runtimes, async (runtime: RuntimeManifest) => {
+          const runtimeDeps = runtime.dependencies as Aspect[];
+          if (runtimeDeps) {
+            const depIds = runtimeDeps.map((d) => d.id).filter((id) => id);
+            const loaded = await this.getManifestsGraphRecursively(depIds, throwOnError);
+            manifests.push(...loaded);
+          }
+        });
+      }
+    });
+    return manifests;
```

Comment:
you might end up with the same manifests multiple times here. not sure how harmony will handle it.
I recommend using uniqBy here to make sure you don't have any duplications

Reply:
I added `uniq` for the ids when I get them. Later, I use the `visited` to make sure I don't have the same id twice.